packages/kit/src/runtime/client/renderer.js

```diff
@@ -384,6 +384,16 @@ export class Renderer {
 					deep_linked.scrollIntoView();
 				} else {
 					scrollTo(0, 0);
+					if (document.body.scrollTop > 0) document.body.scrollTo(0, 0);
+					// In some cases it is needed scroll first root element of target that
+					// is not scrolled to the top
+					for (const target_child of this.target.childNodes) {
+						const target_child_element = target_child.firstChild?.parentElement;
+						if (target_child_element && target_child_element.scrollTop > 0) {
+							target_child_element.scrollTo(0, 0);
+							break;
```

Comment:
I'm not sure it's always safe to `break` here. What if you had one overflowed div inside another?

Reply:
I think I tested this exact usecase and it still worked. Elements that are siblings to the first overflown element don't matter as they cannot overflow more than the first overflown element. Overflowing elements inside the first overflown element also don't matter as they would only scroll inside their parent.
However it could be argued that after a page navigation every scroll position in the DOM should be at 0 because that's the "normal" behavior. But this could be very inefficient if there are many elements in the DOM. That's why I thought breaking as early as possible would be the best solution.